e2e/tests/predicates.rs

```diff
@@ -1245,3 +1245,80 @@ async fn predicate_configurables_in_blobs() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn predicate_transfer_respects_maturity_and_expiration() -> Result<()> {
```

Comment:
At some point, we should discuss how we want to approach testing for features like this, where we cover various transaction and input types. It feels like the level of granularity here (and in other cases) might be higher than what we actually benefit from.